File: odemapster-r2rml-rdb/src/main/java/es/upm/fi/dia/oeg/obdi/wrapper/r2rml/rdb/model/ITemplateTermMap.java
package es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.model;

import java.util.Collection;
import java.util.Map;

public interface ITemplateTermMap {
	public String getTemplateString();
	public Collection<String> getTemplateColumns();
	public Map<String, String> getTemplateValues(String uri);
}

File: morph-r2rml/src/main/scala/es/upm/fi/dia/oeg/morph/r2rml/model/ITemplateTermMap.scala
package es.upm.fi.dia.oeg.morph.r2rml.model

import es.upm.fi.dia.oeg.morph.base.RegexUtility
import scala.collection.JavaConversions._

trait ITemplateTermMap {
	var templateString:String =null;
	
	def getTemplateString():String = { this.templateString };
	
	def getTemplateColumns():Iterable[String] = {
      RegexUtility.getTemplateColumns(this.templateString, true);
   };
	
	def getTemplateValues(uri:String ) : Map[String, String]  = {
		RegexUtility.getTemplateMatching(this.templateString, uri);
	}
}